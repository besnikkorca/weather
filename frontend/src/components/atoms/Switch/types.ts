export interface Props {
  checked: boolean;
  setChecked: (checked: boolean) => void;
}
