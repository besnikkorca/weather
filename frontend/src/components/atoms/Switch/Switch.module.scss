/* The switch - the box around the slider */
.switch {
  position: relative;
  display: inline-block;
  width: 3.75rem;
  height: 2.125rem;

  /* Hide default HTML checkbox */
  & input {
    opacity: 0;
    width: 0;
    height: 0;
  }
  /* The slider */
  & .slider {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: #ccc;
    -webkit-transition: 0.4s;
    transition: 0.4s;
  }

  & .slider:before {
    position: absolute;
    content: '';
    height: 1.625rem;
    width: 1.625rem;
    left: 0.25rem;
    bottom: 0.25rem;
    background-color: white;
    -webkit-transition: 0.4s;
    transition: 0.4s;
  }

  & input:checked + .slider {
    background-color: var(--primary-color);
  }

  & input:focus + .slider {
    box-shadow: 0 0 1px var(--primary-color);
  }

  & input:checked + .slider:before {
    -webkit-transform: translateX(1.625rem);
    -ms-transform: translateX(1.625rem);
    transform: translateX(1.625rem);
  }

  /* Rounded sliders */
  & .slider.round {
    border-radius: 2.125rem;
  }

  & .slider.round:before {
    border-radius: 50%;
  }
}
