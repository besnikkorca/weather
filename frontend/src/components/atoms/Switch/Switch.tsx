import classNames from 'classnames';
import { Component } from 'react';
import styles from './Switch.module.scss';
import { Props } from './types';

export default class Switch extends Component<Props> {
  handleCheck = () => {
    const { checked, setChecked } = this.props;
    if (!setChecked) return;

    setChecked(!checked);
  };

  render() {
    const { checked } = this.props;
    return (
      <label className={styles.switch}>
        <input type="checkbox" checked={checked} onChange={this.handleCheck} />
        <span className={classNames(styles.slider, styles.round)}></span>
      </label>
    );
  }
}
