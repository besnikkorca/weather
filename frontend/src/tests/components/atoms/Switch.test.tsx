import { render, screen } from '@testing-library/react';
import Switch from 'atoms/Switch';

describe('Switch', () => {
  it('should render correctly', () => {
    const { container } = render(<Switch checked setChecked={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  it('should render correctly when not checked', () => {
    const { container } = render(<Switch checked={false} setChecked={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  it('should render correctly when checked', () => {
    const { container } = render(<Switch checked setChecked={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  it('should call setChecked when clicked', () => {
    const setChecked = jest.fn();
    render(<Switch checked setChecked={setChecked} />);
    screen.getByRole('checkbox').click();
    expect(setChecked).toHaveBeenCalled();
  });

  it('should call setChecked with correct value when clicked', () => {
    const setChecked = jest.fn();
    render(<Switch checked setChecked={setChecked} />);
    screen.getByRole('checkbox').click();
    expect(setChecked).toHaveBeenCalledWith(false);
  });
});
