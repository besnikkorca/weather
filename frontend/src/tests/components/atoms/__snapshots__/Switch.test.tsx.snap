// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should render correctly 1`] = `
<div>
  <label
    class="switch"
  >
    <input
      checked=""
      type="checkbox"
    />
    <span
      class="slider round"
    />
  </label>
</div>
`;

exports[`Switch should render correctly when checked 1`] = `
<div>
  <label
    class="switch"
  >
    <input
      checked=""
      type="checkbox"
    />
    <span
      class="slider round"
    />
  </label>
</div>
`;

exports[`Switch should render correctly when not checked 1`] = `
<div>
  <label
    class="switch"
  >
    <input
      type="checkbox"
    />
    <span
      class="slider round"
    />
  </label>
</div>
`;
